spec/abstract-jobs.html

```diff
@@ -130,25 +130,19 @@ <h1>Liveness</h1>
     <emu-clause id="sec-weakref-execution">
       <h1>Execution</h1>
 
-      <p>
-        At any time, an ECMAScript implementation may perform any of the three
-        following actions. They may be triggered interspersed with ECMAScript
-        code execution, or, e.g., from a separate parallel thread.
-      </p>
-
-      <ul>
-        <li>
-          If an object is not live, then the implementation may replace any
-          reference to the object from the [[Target]] field of a
-          FinalizationGroup's [[Cells]] List, or the [[Target]] internal slot
-          of a WeakRef, to the value ~empty~.
-        </li>
-        <li>
-          When there is a FinalizationGroup _fg_ such that
-          CheckForEmptyCells(_fg_) is *true*, the implementation may perform
-          HostCleanupFinalizationGroup(_fg_).
-        </li>
-      </ul>
+        At any time, if an object _obj_ is not live, an ECMAScript implementation may perform the following steps atomically:
+
+      <emu-alg>
+        1. For each WeakRef _ref_ such that _ref_.[[Target]] is _obj_,
```

Comment:
Does this automatically apply to all realms in all agents, or should they be explicitly iterated?

Reply:
There are no cross-agent references. I don't think we have a concept of an object being "in a Realm", so I'm not sure what it would mean to enumerate them.